src/clients/FlexbaseClient.Card.ts

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Card } from '../models/Card/Card';
+import { Address } from '../models/Address/Address';
+import { CardOptions } from '../models/Card/CardOptions';
+import { FlexbaseClientBase } from './FlexbaseClient.Base';
+import { FlexbaseResponse } from '../models/FlexbaseResponse';
+
+interface QueryParameters {
+    searchTerm?: string;
+    status?: string;
+}
+
+interface CardsResponse extends FlexbaseResponse{
+    cards: Card[];
+}
+
+interface CardResponse extends FlexbaseResponse{
+    card: Card;
+}
+
+interface IssueCardForm {
+    cardType: string,
+    shipTo?: Address,
+    service?: string,
+}
+
+interface UpdateCardForm {
+    expensesTypes: {
+        amount: number,
+        groups: any,
+    };
+    notifyUse: boolean;
+    shipTo?: Address,
+    creditLimit: number,
+    cardName?: string,
+}
+
+export class FlexbaseClientCard extends FlexbaseClientBase {
+    private params({ searchTerm, status }: CardOptions = {}) {
+        const params: any = {};
+
+        if (typeof searchTerm === 'string') {
+            params.searchTerm = searchTerm;
+          }
+      
+          if (typeof status === 'string') {
+            params.status = status;
+          }
+
+        return params;
+    }
+
+    async getCompanyCards({ searchTerm, status }: QueryParameters = {}): Promise<Card[] | null> {
+        try {
+          const params = this.params({ searchTerm, status });
+    
+          const response = await this.client.url('/card/company').query(params).get().json<CardsResponse>();
+
+           if (!response.success) {
+              return null;
+           }
+    
+          return response.cards;
+        } catch (error) {
+          this.logger.error('Unable to get company cards', error);
+          return null;
+        }
+    }
+
+    async getUserCard(userId: string): Promise<Card | null> {
+        try {    
+          const response = await this.client.url(`/card/${userId}`).get().json<CardResponse>();
+
+           if (!response.success) {
+              return null;
+           }
+    
+          return response.card;
+        } catch (error) {
+          this.logger.error('Unable to get user card', error);
+          return null;
```

Comment:
See comment above. We need to determine whether returning `null` or a contextual message around the error is the best course of action here.

Reply:
@robert-dean 
On second thought, this is probably not the route we want to take here. For one, incorporating OOP paradigms in our functional programming-heavy repo will add a ton of unnecessary layers of abstraction that will only create headaches for us in the future. Secondly, using generics here will add lots of additional boilerplate code throughout the repo, causing us to undertake a massive refactoring effort. Essentially, in this case, we'd have to refactor all of our response objects on both the client side as well as the frontend. We need to keep our code as lean and navigable as possible. 

Hence, instead of returning a standardized wrapper object here, I'm suggesting that we just return a generic error message. As you mentioned in another comment above, we should adhere to best security practices, so the more generic the message, the better. Looping in @kyle-b2 and @peilong-du for ideas/reservations as well

Reply:
Yeah, one of the benefits of javascript is that it's very flexible. We can easily change the errors should we need to, with minimal effort. I'm not a fan of all the layers and layers of abstraction in javascript. It makes sense for an OOP language like Java/C#, but less so in javascript.

Maybe it's because I wasn't in the java world for more than 3 years or so, but I've only ever seen interfaces and generics justified as good 'in case we need to swap something out', but I've never actually seen people do it. It usually just ends up being more overhead with no implementation swap.

If we had all the time in the world, I'd like to make everything as generic and re-useable as possible, but I don't think it's worth the trade off given how fast we need to move.